### ISSUE TITLE
MacOS: Error while compiling 

### ISSUE THREAD

**Issue by aliakseikalosha** [mmowbray/clion-modern-opengl#1]
I have downloaded and install all libraries with brew but still getting this error while copiling this project.
```
====================[ Build | clion_modern_opengl | Debug ]=====================
/Applications/CLion.app/Contents/bin/cmake/mac/bin/cmake --build /Users/aliakseikalosha/Documents/git/work/AllTaskFromTUL/TUL/20212022/PG2/clion-modern-opengl-master/cmake-build-debug --target clion_modern_opengl -- -j 9
Consolidate compiler generated dependencies of target clion_modern_opengl
[ 33%] Linking CXX executable clion_modern_opengl
Undefined symbols for architecture x86_64:
  "___glewAttachShader", referenced from:
      GLSLProgram::compileShaderFromString(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int) in GLSLProgram.cpp.o
  "___glewBindAttribLocation", referenced from:
      GLSLProgram::bindAttribLocation(unsigned int, char const*) in GLSLProgram.cpp.o
  "___glewBindBuffer", referenced from:
      _main in App.cpp.o
  "___glewBindFragDataLocation", referenced from:
      GLSLProgram::bindFragDataLocation(unsigned int, char const*) in GLSLProgram.cpp.o
  "___glewBindVertexArray", referenced from:
      _main in App.cpp.o
  "___glewBufferData", referenced from:
      _main in App.cpp.o
  "___glewCompileShader", referenced from:
      GLSLProgram::compileShaderFromString(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int) in GLSLProgram.cpp.o
  "___glewCreateProgram", referenced from:
      GLSLProgram::GLSLProgram() in GLSLProgram.cpp.o
  "___glewCreateShader", referenced from:
      GLSLProgram::compileShaderFromString(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int) in GLSLProgram.cpp.o
  "___glewDeleteProgram", referenced from:
      GLSLProgram::link() in GLSLProgram.cpp.o
  "___glewEnableVertexAttribArray", referenced from:
      _main in App.cpp.o
  "___glewGenBuffers", referenced from:
      _main in App.cpp.o
  "___glewGenVertexArrays", referenced from:
      _main in App.cpp.o
  "___glewGetActiveAttrib", referenced from:
      GLSLProgram::printActiveAttribs() in GLSLProgram.cpp.o
  "___glewGetActiveUniform", referenced from:
      GLSLProgram::printActiveUniforms() in GLSLProgram.cpp.o
  "___glewGetProgramiv", referenced from:
      GLSLProgram::link() in GLSLProgram.cpp.o
      GLSLProgram::printActiveUniforms() in GLSLProgram.cpp.o
      GLSLProgram::printActiveAttribs() in GLSLProgram.cpp.o
  "___glewGetShaderInfoLog", referenced from:
      GLSLProgram::compileShaderFromString(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int) in GLSLProgram.cpp.o
      GLSLProgram::link() in GLSLProgram.cpp.o
  "___glewGetShaderiv", referenced from:
      GLSLProgram::compileShaderFromString(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int) in GLSLProgram.cpp.o
  "___glewGetUniformLocation", referenced from:
      GLSLProgram::getUniformLocation(char const*) in GLSLProgram.cpp.o
  "___glewLinkProgram", referenced from:
      GLSLProgram::link() in GLSLProgram.cpp.o
  "___glewShaderSource", referenced from:
      GLSLProgram::compileShaderFromString(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int) in GLSLProgram.cpp.o
  "___glewUniform1f", referenced from:
      GLSLProgram::setUniform(char const*, float) in GLSLProgram.cpp.o
  "___glewUniform1i", referenced from:
      GLSLProgram::setUniform(char const*, int) in GLSLProgram.cpp.o
      GLSLProgram::setUniform(char const*, bool) in GLSLProgram.cpp.o
  "___glewUniform3f", referenced from:
      GLSLProgram::setUniform(char const*, float, float, float) in GLSLProgram.cpp.o
  "___glewUniform3fv", referenced from:
      GLSLProgram::setUniform(char const*, glm::vec<3, float, (glm::qualifier)0> const&) in GLSLProgram.cpp.o
  "___glewUniform4fv", referenced from:
      GLSLProgram::setUniform(char const*, glm::vec<4, float, (glm::qualifier)0> const&) in GLSLProgram.cpp.o
  "___glewUniformMatrix3fv", referenced from:
      GLSLProgram::setUniform(char const*, glm::mat<3, 3, float, (glm::qualifier)0> const&) in GLSLProgram.cpp.o
  "___glewUniformMatrix4fv", referenced from:
      GLSLProgram::setUniform(char const*, glm::mat<4, 4, float, (glm::qualifier)0> const&) in GLSLProgram.cpp.o
  "___glewUseProgram", referenced from:
      GLSLProgram::use() in GLSLProgram.cpp.o
  "___glewVertexAttribPointer", referenced from:
      _main in App.cpp.o
  "_glewExperimental", referenced from:
      _main in App.cpp.o
  "_glewInit", referenced from:
      _main in App.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [clion_modern_opengl] Error 1
make[2]: *** [CMakeFiles/clion_modern_opengl.dir/all] Error 2
make[1]: *** [CMakeFiles/clion_modern_opengl.dir/rule] Error 2
make: *** [clion_modern_opengl] Error 2

```

So as I understand it is because clion trying to use dynamic libraries and mac os whan to use static variant, so the question is how to setup cmake that, so it link to static libraries?

**Comment by aliakseikalosha** [comment]
So to fix it you need to remove any mentions of GLEW. It will work without GLEW.

**Comment by aliakseikalosha** [comment]
this issue is solved by #2
